### DISCUSSION TITLE
Processing ERA5 data in LVT

### DISCUSSION THREAD

**Discussion by slinskik** [NASA-LIS/LISF#1089] [Q&A] [answered]
Hi LIS team - I am encountering errors when I try to process ERA5 precipitation and surface air temperature data in LVT.  The error I get is:

****************Error****************************
observation init routine for source ERA5 is not defined
please see the configs/lvt.config.master file.....
program will seg fault.....
****************Error****************************

Any advice would be greatly appreciated.  I have attached my config file.  Many thanks for your help!

[lvt_config_2021.TXT](https://github.com/NASA-LIS/LISF/files/8672909/lvt_config_2021.TXT)

.

**Comment by bmcandr** [comment]
Hi @slinskik,

I went back and looked at pull request #646 that added this feature and I noticed the same thing during my review. In that thread @sujayvkumar indicated that this is expected behavior so he may be able to explain what's going on.

**Comment by slinskik** [comment, marked as answer]
@bmcandr  - many thanks for the pull request information!  

I was able to process ERA5 data after taking the following steps:

1) downloaded and compiled the latest LVT executable from the master branch
2) added the following constraint to my slurm file: #SBATCH --constraint="sky|cas" (ERA5 processing in LVT crashes when run on a compute node and on Haswell nodes - this pertains only to running on Discover)

Also, I needed to correct the datastream attributes in my lvt.config file.  They are:
TotalPrecip    1 1  kg/m2  DN  3 1
Tair_f              1 1  K          -      1 1

Many thanks to @emkemp for his help troubleshooting this and finding a solution.
